### PR TITLE
Update RSPECs; deprecate S4784

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
note that I ignored the deprecation of RSPEC-4784 - that was replaced on Java side by other rules which we don't have yet

### CODE REVIEW

File: sonaranalyzer-dotnet/rspec/cs/S1313_c#.html [hunk 1]
```diff
@@ -28,9 +33,13 @@ <h2>Compliant Solution</h2>
 var address = IPAddress.Parse(ip);
 </pre>
 <h2>Exceptions</h2>
+<p>No issue is reported for the following cases because they are not considered sensitive:</p>
 <ul>
-  <li> Although "::" is a valid IPv6 address, the rule doesn't report on it. </li>
-  <li> No issue is reported for 127.0.0.1 because loopback is not considered as sensitive </li>
+  <li> Loopback addresses 127.0.0.0/8 in CIDR notation (from 127.0.0.0 to 127.255.255.255) </li>
+  <li> Broadcast address 255.255.255.255 </li>
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/rspec/cs/S1313_c#.html:39]: Was this implemented on our side?

File: sonaranalyzer-dotnet/rspec/cs/S5547_c#.html [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
+<p>For <a href="https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography?view=netcore-3.1">System.Security.Cryptography</a> library,
+these old cryptographic algorithms should no longer be used for any reason:</p>
 <pre>
 var tripleDES1 = new TripleDESCryptoServiceProvider(); // Noncompliant: Triple DES is vulnerable to meet-in-the-middle attack
 
 var simpleDES = new DESCryptoServiceProvider(); // Noncompliant: DES works with 56-bit keys allow attacks via exhaustive search
 
 var RC2 = new RC2CryptoServiceProvider(); // Noncompliant: RC2 is vulnerable to a related-key attack
 </pre>
+<p>For Bouncycastle library, <a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-1000339">AESFastEngine has a side channel leak</a>, it
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/rspec/cs/S5547_c#.html:15]: I'm not sure if this is tied to new functionality, I'll ask @eric-therond-sonarsource

File: sonaranalyzer-dotnet/rspec/cs/S1313_c#.html (hunk 1 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/rspec/cs/S1313_c#.html:39]: I'll look

File: sonaranalyzer-dotnet/rspec/cs/S5547_c#.html (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/rspec/cs/S5547_c#.html:15]: From what I can see in the code only https://rules.sonarsource.com/csharp/RSPEC-4426 has checks for Bouncycastle types.

File: sonaranalyzer-dotnet/rspec/vbnet/S1313_vb.net.html [hunk 3]
```diff
@@ -28,9 +33,13 @@ <h2>Compliant Solution</h2>
 Dim address = IPAddress.Parse(ip)
 </pre>
 <h2>Exceptions</h2>
+<p>No issue is reported for the following cases because they are not considered sensitive:</p>
 <ul>
-  <li> Although "::" is a valid IPv6 address, the rule doesn't report on it. </li>
-  <li> No issue is reported for 127.0.0.1 because loopback is not considered as sensitive </li>
+  <li> Loopback addresses 127.0.0.0/8 in CIDR notation (from 127.0.0.0 to 127.255.255.255) </li>
+  <li> Broadcast address 255.255.255.255 </li>
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/rspec/vbnet/S1313_vb.net.html:39]: I could not find any checks for "255.255.255.255" or "0.0.0.0". I think we need to add a jira ticket for that.

@eric-therond-sonarsource: do we have an MMF for this?

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S1313_cs.html [hunk 4]
```diff
@@ -28,9 +33,13 @@ <h2>Compliant Solution</h2>
 var address = IPAddress.Parse(ip);
 </pre>
 <h2>Exceptions</h2>
+<p>No issue is reported for the following cases because they are not considered sensitive:</p>
 <ul>
-  <li> Although "::" is a valid IPv6 address, the rule doesn't report on it. </li>
-  <li> No issue is reported for 127.0.0.1 because loopback is not considered as sensitive </li>
+  <li> Loopback addresses 127.0.0.0/8 in CIDR notation (from 127.0.0.0 to 127.255.255.255) </li>
+  <li> Broadcast address 255.255.255.255 </li>
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S1313_cs.html:39]: same comment here

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S1313_vb.html [hunk 5]
```diff
@@ -28,9 +33,13 @@ <h2>Compliant Solution</h2>
 Dim address = IPAddress.Parse(ip)
 </pre>
 <h2>Exceptions</h2>
+<p>No issue is reported for the following cases because they are not considered sensitive:</p>
 <ul>
-  <li> Although "::" is a valid IPv6 address, the rule doesn't report on it. </li>
-  <li> No issue is reported for 127.0.0.1 because loopback is not considered as sensitive </li>
+  <li> Loopback addresses 127.0.0.0/8 in CIDR notation (from 127.0.0.0 to 127.255.255.255) </li>
+  <li> Broadcast address 255.255.255.255 </li>
+  <li> Non routable address 0.0.0.0 </li>
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S1313_vb.html:40]: and here

**Comment by andrei-epure-sonarsource** [comment]
@costin-zaharia-sonarsource I reverted the S1313 and S5547 changes

File: sonaranalyzer-dotnet/rspec/cs/S5547_c#.html (hunk 2 above)
Comment (eric-therond-sonarsource) [reply sonaranalyzer-dotnet/rspec/cs/S5547_c#.html]: these changes are related to this ticket: https://github.com/SonarSource/sonar-dotnet/issues/3330

File: sonaranalyzer-dotnet/rspec/vbnet/S1313_vb.net.html (hunk 3 above)
Comment (eric-therond-sonarsource) [reply sonaranalyzer-dotnet/rspec/vbnet/S1313_vb.net.html]: I just created this ticket to fix the gap: https://github.com/SonarSource/sonar-dotnet/issues/3450

**Comment by eric-therond-sonarsource** [comment]
RSPEC-4784 has been deprecated for all languages on purposes.
Independently of the work that was done in Java about regular expressions.
In the appsec bubble we consider this rule too noisy and could be deprecated right now to provide a better security-hotspots experience to the users.

**Pull request merged**